x-pack/plugins/cloud_security_posture/public/plugin.tsx

```diff
@@ -56,6 +56,11 @@ export class CspPlugin
   }
 
   public start(core: CoreStart, plugins: CspClientPluginStartDeps): CspClientPluginStart {
+    plugins.data.dataViews.createAndSave({
```

Comment:
Right now we have 4 data views installed by our plugin, eg. https://github.com/elastic/integrations/blob/main/packages/cloud_security_posture/kibana/index_pattern/cloud_security_posture-07a5e6d6-982d-4c7c-a845-5f2be43279c9.json First of all, am I correct that these are Data Views and actually Index Patterns as per the terminology you used in this ticket? I think the confusion comes from the fact that in Kibana historically Data Views were named Index Patterns and this naming is kept in some places even now. If yes, the question is:
- do we need two data views per finding type (2 for misconfiguration and 2 for vulnerability). Do we even use the "data stream" data views anywhere?
- not sure if it's a part of you task but I think we might need to think about migration strategy to decommission the data views inside our package (if it's possible ofc)

Reply:
> I think the confusion comes from the fact that in Kibana historically Data Views were named Index Patterns 

Yes, you are correct here. The presence of the index pattern in the data view is another reason for the confusion.

>do we need two data views per finding type (2 for misconfiguration and 2 for vulnerability). Do we even use the "data stream" data views anywhere?

I am pretty sure that we need only the "latest" data view. I will check that when I deprecate the integration's data views. ([here](https://github.com/elastic/kibana/issues/187226))

> not sure if it's a part of you task but I think we might need to think about migration strategy to decommission the data views inside our package (if it's possible ofc)

In ESS, this isn't an issue because we can synchronize the integration version with the Kibana version. 
This ensures that we won't deprecate the data views from the integration without linking it to a Kibana version that includes the new data view.

For serverless, you're correct. I'm currently [working](https://github.com/elastic/security-team/issues/9666) on introducing the new data view without replacing it yet. This way, it will be available in the latest serverless version before we fully transition to the new data view. This approach seems like a solid plan.